fuzz/fuzz_targets/fs.rs

```diff
@@ -0,0 +1,230 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+//
+// Copyright (c) 2023 SUSE LLC
+//
+// Author: Carlos López <carlos.lopez@suse.com>
+
+#![no_main]
+
+use arbitrary::Arbitrary;
+use core::hint::black_box;
+use libfuzzer_sys::fuzz_target;
+use std::rc::Rc;
+use std::sync::OnceLock;
+use svsm::fs::FileHandle;
+use svsm::fs::{create, create_all, initialize_fs, list_dir, mkdir, open, uninitialize_fs, unlink};
+use svsm::mm::alloc::TestRootMem;
+
+const ROOT_MEM_SIZE: usize = 0x10000;
+const MAX_READ_SIZE: usize = 4096 * 8;
+const MAX_WRITE_SIZE: usize = 4096 * 8;
+const WRITE_BYTE: u8 = 0x0;
+const POISON_BYTE: u8 = 0xaf;
+
+#[derive(Arbitrary, Debug)]
+enum FsAction<'a> {
+    Create(&'a str),
+    CreateNamed(usize),
+    CreateAll(&'a str),
+    CreateAllNamed(usize),
+    Open(&'a str),
+    OpenNamed(usize),
+    Close(usize),
+    Unlink(&'a str),
+    UnlinkNamed(usize),
+    Mkdir(&'a str),
+    MkdirNamed(usize),
+    ListDir(&'a str),
+    ListDirNamed(usize),
+    Read(usize, usize),
+    Write(usize, usize),
+    Seek(usize, usize),
+    Truncate(usize, usize),
+}
+
+fn get_idx<T>(v: &[T], idx: usize) -> Option<usize> {
+    if !v.is_empty() {
+        Some(idx % v.len())
+    } else {
+        None
+    }
+}
+
+/// A handle for a file that also holds its original name
+#[derive(Debug)]
+struct Handle {
+    fd: FileHandle,
+    name: Rc<str>,
+}
```

Comment:
```suggestion
/// A handle for a file that also holds its original name
#[derive(Debug)]
struct Handle<'a> {
    fd: FileHandle,
    name: &'a str,
}
```

Allocating into an `Rc` is not needed here.